src/Var.cc

```diff
@@ -223,6 +223,12 @@ static void make_var(const IDPtr& id, TypePtr t, InitClass c, ExprPtr init,
 				id->Warn("redefinition requires \"redef\"", redef_obj, true);
 			}
 
+		else if ( dt == VAR_REDEF && ! id->IsRedefinable() )
```

Comment:
Not that I know my way around here, but wondering if the `if ( dt == VAR_REDEF )` condition of the block below could be re-used?

https://github.com/zeek/zeek/pull/2295/files#diff-7b1d693024302c91261b7aae6d2a1d658971067a46b912c3965b44f0fd4f1663R243

Reply:
I had added it there originally but it falls into the block directly below the one I added. Since we're redef'ing a function in this instance, it ends up calling `add_prototype` and gets into the call chain that eventually results in the crash. Adding the extra if block avoids that.